python/pyspark/testing/utils.py

```diff
@@ -592,7 +616,7 @@ def assert_rows_equal(rows1: List[Row], rows2: List[Row]):
             rows_str2 += str(r2) + "\n"
             if not compare_rows(r1, r2):
                 diff_rows_cnt += 1
-                diff_rows = True
+                diff_rows.append((r1, r2))
```

Comment:
@itholic let's optimize this. if `includeDiffRows` is disabled, we don't need to keep the diff rows which costs memory.

Reply:
Sounds good. Updated